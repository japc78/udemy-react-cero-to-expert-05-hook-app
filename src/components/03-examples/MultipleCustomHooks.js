import React from 'react'
import { useCounter } from '../../hooks/useCounter'
import { useFetch } from '../../hooks/useFetch'
import './styles.css'
export const MultipleCustomHooks = () => {

	const { counter, increment } = useCounter(1);

	// const state = useFecth('https://www.breakingbadapi.com/api/quotes/1');
	const { loading, data } = useFetch(`https://www.breakingbadapi.com/api/quotes/${counter}`);

	//doble negacion !! esplicacion: !data, sino tiene valor, !!data == data, es decir si tiene valor.
	// https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Operadores/Operadores_l%C3%B3gicos
	const { author, quote } = !!data && data[0];

	// console.log(loading);

	return (
		<>
			<h1>BreakingBad Quotes</h1>
			<hr/>

			{
				loading
				?
				(<div className="alert alert-info">loading...</div>)
				:
				(
					<blockquote className="blockquote text-right">
						<p className="mb-0">{quote}</p>
						<footer className="blockquote-footer">{author}</footer>
					</blockquote>
				)
			}

			<button className="btn btn-primary" onClick={ increment }>Siguiente frase</button>
		</>
	)
}
