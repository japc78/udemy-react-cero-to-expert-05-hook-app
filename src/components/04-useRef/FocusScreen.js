import React, { useRef } from 'react'
import './styles.css'
export const FocusScreen = () => {

	const inputRef = useRef();
	// console.log(inputRef);

	const handleClick = () => {
		// document.querySelector('input').focus();
		// document.querySelector('input').select();

		// Con useRef
		inputRef.current.select();

	}

	return (
		<>
			<h1>FocusScreen</h1>
			<hr/>

			<div className="mb-3">
				<input
					ref = { inputRef }
					type="text"
					className="form-contol"
					placeholder="Su nombre"
				/>
			</div>
			<button onClick={ handleClick } className="btn btn-sm btn-outline-primary">Focus</button>
		</>
	)
}
