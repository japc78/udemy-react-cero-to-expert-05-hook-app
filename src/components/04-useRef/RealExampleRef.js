import React, { useRef, useState } from 'react'
import { MultipleCustomHooks } from '../03-examples/MultipleCustomHooks'
import './styles.css'
export const RealExampleRef = () => {

	const [show, setShow] = useState(false);
	const change = () => {
		//Cambia show al valor opuesto
		setShow(!show)
	}



	return (
		<>
			<h1>RealExampleRef</h1>
			<hr/>

			{/* Si show esta en true muestra el componente, sino lo oculta */}
			{show && <MultipleCustomHooks />}

			<button
				className="btn btn-sm btn-outline-primary d-block mt-3"
				onClick = { change }
			>Show & Hide</button>

		</>
	)
}
