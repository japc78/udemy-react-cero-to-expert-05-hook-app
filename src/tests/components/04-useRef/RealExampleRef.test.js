import '@testing-library/jest-dom'
import { shallow } from 'enzyme';
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { RealExampleRef } from '../../../components/04-useRef/RealExampleRef';

describe('Test interactions with the useState', () => {
	const wrapper = shallow(<RealExampleRef />);

	test('should be shown correctly ', () => {
		expect(wrapper).toMatchSnapshot();
		expect( wrapper.find('MultipleCustomHooks').exists()).toBe(false);
	});


	test('should be shown the component <MultipleCustomHooks />', () => {
		wrapper.find('button').simulate('click');
		expect( wrapper.find('MultipleCustomHooks').exists()).toBe(true);
	});
})
