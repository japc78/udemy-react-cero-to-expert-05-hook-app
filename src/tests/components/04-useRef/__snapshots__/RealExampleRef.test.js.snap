// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test interactions with the useState should be shown correctly  1`] = `
<Fragment>
  <h1>
    RealExampleRef
  </h1>
  <hr />
  <button
    className="btn btn-sm btn-outline-primary d-block mt-3"
    onClick={[Function]}
  >
    Show & Hide
  </button>
</Fragment>
`;
